### ISSUE TITLE
v1.12 doesn't work with 4.7 .net project. Some runtime serialization code not available.

### ISSUE THREAD

**Issue by DonWM** [lppkarl/YahooFinanceApi#11]
Cannot use v 1.12 in a 4.7 .net project.  Gives a message that some runtime serialization code it not available.

Might have something to do with the Name function in Helper and/or the slimsemaphore

**Comment by lppkarl** [comment]
@DonWM Is that the "System.Runtime.Serialization.Primitives" missing error? You may refer to #8 

Nuget doesn't include this reference when installing this library, and it's only happens to the traditional .net framework project but not .net core project. Manually install the reference should work

**Comment by DonWM** [comment]
I didn’t see the “System.Runtime.Serialization.Primitives” as a reference to be installed.  Where do I find it?

 

By the way, thanks for doing all this.  Your code is VERY nice!

 

Thanks,

Don

**Comment by lppkarl** [comment]
@DonWM Thanks for the praise :) For the "System.Runtime.Serialization.Primitives" package, you can pull it directly from nuget just like what you did with this lib

**Comment by lppkarl** [comment]
This issue has been idled for a month, it will be closed now

**Comment by DonWM** [comment]
Karl,

 

I attempted to upgrade to v2.0 from v1.12 and it wouldn’t allow update with my 4.7 .net target vs. 4.6.1 target even though you said it should work for 4.6.1 or above. 

 

I also tried to download and load the YahooFinanceApi solution and it had errors.  It said it needed the netstandard v2.0 in the YahooFinanceApi project but I saw the reference in that project to only to v1.1.0 and not v2.0.

 

What am I missing and how can I update to your new v2.0?

 

Also last week on Tue and Weds, I got the “access denied error 999” error in the YahooQuote method, so couldn’t get any stock quotes.  Do you know what the issue may be?

 

I didn’t try it on Thursday or Friday.

 

Thanks,

Don Maurer

**Comment by lppkarl** [comment]
Hi Don,

You may want try completely uninstall the YahooFinanceApi package & install the latest version again, 4.7 should be ok for net standard 2.0 library: https://github.com/dotnet/announcements/issues/32 <https://github.com/dotnet/announcements/issues/32> . 

For the access denied issue, there is an opened issue here: https://github.com/lppkarl/YahooFinanceApi/issues/23 <https://github.com/lppkarl/YahooFinanceApi/issues/23> . The main reason for the issue is that Yahoo seems to terminate the quote service, and it affects the other libraries relying on the service. I’m keeping an eye on it, and hopefully try to find a way to workaround it.

Regards,
Karl

**Comment by DonWM** [comment]
Severity

Code

Description

Project

File

Line

Suppression State


Error

 

Could not install package 'YahooFinanceApi 2.0.0'. You are trying to install this package into a project that targets '.NETFramework,Version=v4.7.1', but the package does not contain any assembly references or content files that are compatible with that framework. For more information, contact the package author.

 

 

0

 

 

Karl,

 

I got the above error when I uninstalled v.1.12 and then tried to install v.2.0.  Your package is not liking either v4.7 or v4.7.1 which I installed hoping it would help.

 

I also downloaded and installed .net core v2.0.2 sdk.

 

By the way, other users of other interfaces to Yahoo had the same denied access error as I did for current Yahoo quotes, so maybe they are changing  and maybe starting to use a “crumb” or something for that interface to slow down programmatic access to the data.

 

Don
